src/common/hist_util.cc

```diff
@@ -785,41 +860,46 @@ void BuildHistSparseKernel(const common::Span<const size_t> rid_span, const floa
       const size_t icol_end_prefect = row_ptr[rid[i+Prefetch::kPrefetchOffset]+1];
 
       PREFETCH_READ_T0(pgh + 2*rid[i + Prefetch::kPrefetchOffset]);
-      for (size_t j = icol_start_prftch; j < icol_end_prefect; j+=Prefetch::kPrefetchStep) {
+      for (size_t j = icol_start_prftch; j < icol_end_prefect; j+=prefetch_step) {
```

Comment:
I think you can keep Prefetch::kPrefetchStep as is.

Reply:
As I correctly understand  Prefetch::kPrefetchStep depends on size if index. So template function was added for it:
template <typename T>
size_t getPrefetchStep()

Reply:
You can write like this, if you make GetPrefetchStep as constexpr function
```cpp
for (size_t j = icol_start_prftch; j < icol_end_prefect; j+=GetPrefetchStep<T>()) {
```

Reply:
done